packages/pangea-sdk/pangea/services/authn/models.py

```diff
@@ -434,281 +287,205 @@ class ClientJWKSResult(PangeaResponseResult):
-    flow_id: str
-    password: str
-    first_name: str
-    last_name: str
+class FlowRestartDataSMSOTP(APIRequestModel):
+    phone: str
 
 
-class FlowSignupPasswordResult(CommonFlowResult):
-    pass
+FlowRestartData = Union[Dict, FlowRestartDataSMSOTP]
 
 
-# - path: authn::/v1/flow/signup/social
-# https://pangea.cloud/docs/api/authn#social-sign-up
-class FlowSignupSocialRequest(APIRequestModel):
+class FlowRestartRequest(APIRequestModel):
     flow_id: str
-    cb_state: str
-    cb_code: str
+    choice: FlowChoice
+    data: FlowRestartData
 
 
-class FlowSignupSocialResult(CommonFlowResult):
+class FlowRestartResult(CommonFlowResult):
     pass
 
 
-# - path: authn::/v1/flow/start
-# https://pangea.cloud/docs/api/authn#start-a-sign-up-in
 class FlowStartRequest(APIRequestModel):
     cb_uri: Optional[str] = None
     email: Optional[str] = None
     flow_types: Optional[List[FlowType]] = None
-    provider: Optional[IDProvider] = None
     invitation: Optional[str] = None
 
 
 class FlowStartResult(CommonFlowResult):
     pass
 
 
-# - path: authn::/v1/flow/verify/captcha
-# https://pangea.cloud/docs/api/authn#verify-captcha
-class FlowVerifyCaptchaRequest(APIRequestModel):
-    flow_id: str
-    code: str
+class FlowUpdateDataAgreements(APIRequestModel):
+    agreed: List[str]
 
 
-class FlowVerifyCaptchaResult(CommonFlowResult):
-    pass
-
-
-# - path: authn::/v1/flow/verify/email
-# https://pangea.cloud/docs/api/authn#verify-email-address
-class FlowVerifyEmailRequest(APIRequestModel):
-    flow_id: str
-    cb_state: Optional[str] = None
-    cb_code: Optional[str] = None
-
-
-class FlowVerifyEmailResult(CommonFlowResult):
-    pass
-
-
-# - path: authn::/v1/flow/verify/mfa/complete
-# https://pangea.cloud/docs/api/authn#complete-mfa-verification
-class FlowVerifyMFACompleteRequest(APIRequestModel):
-    flow_id: str
-    code: Optional[str] = None
-    cancel: Optional[bool] = None
+class FlowUpdateDataCaptcha(APIRequestModel):
+    code: str
 
 
-class FlowVerifyMFACompleteResult(CommonFlowResult):
-    pass
+class FlowUpdateDataEmailOTP(APIResponseModel):
+    code: str
 
 
-# - path: authn::/v1/flow/verify/mfa/start
-# https://pangea.cloud/docs/api/authn#start-mfa-verification
-class FlowVerifyMFAStartRequest(APIRequestModel):
-    flow_id: str
-    mfa_provider: MFAProvider
+class FlowUpdateDataMagiclink(APIRequestModel):
+    state: str
+    code: str
 
 
-class FlowVerifyMFAStartResult(CommonFlowResult):
+class FlowUpdateDataPasskey(APIRequestModel):
+    # FIXME: is empty?
     pass
 
 
-# - path: authn::/v1/flow/verify/password
-# https://pangea.cloud/docs/api/authn#password-sign-in
-class FlowVerifyPasswordRequest(APIRequestModel):
-    flow_id: str
-    password: Optional[str] = None
-    cancel: Optional[bool] = None
+class FlowUpdateDataPassword(APIRequestModel):
+    password: str
 
 
-class FlowVerifyPasswordResult(CommonFlowResult):
-    pass
+class FlowUpdateDataProfile(APIRequestModel):
+    profile: Profile
```

Comment:
this needs to support providing other fields besides first/last/phone.

Reply:
It does, it derivate of a dict so you can add other fields too